pkg/services/sqlstore/annotation.go

```diff
@@ -239,6 +256,38 @@ func (r *SQLAnnotationRepo) Find(ctx context.Context, query *annotations.ItemQue
 	return items, err
 }
 
+func getAccessControlFilter(user *models.SignedInUser) (string, []interface{}, error) {
+	if user == nil || user.Permissions == nil || user.Permissions[user.OrgId] == nil {
+		return "", nil, errors.New("missing permissions")
+	}
+	if _, has := user.Permissions[user.OrgId][ac.ActionAnnotationsRead]; !has {
+		return "", nil, errors.New("missing permissions")
+	}
+
+	scopes := user.Permissions[user.OrgId][ac.ActionAnnotationsRead]
+	types, hasWildcardScope := ac.ParseScopes(ac.ScopeAnnotationsProvider.GetResourceScopeType(""), scopes)
+	if hasWildcardScope {
+		types = map[interface{}]struct{}{annotations.Dashboard.String(): {}, annotations.Organization.String(): {}}
+	}
+
+	var filters []string
+	var params []interface{}
+	for t := range types {
+		// annotation read permission with scope annotations:type:organization allows listing annotations that are not associated with a dashboard
+		if t == annotations.Organization.String() {
+			filters = append(filters, "a.dashboard_id = 0")
+		}
+		// annotation read permission with scope annotations:type:dashboard allows listing annotations from dashboards which the user can view
+		if t == annotations.Dashboard.String() {
+			dashboardFilter, dashboardParams := permissions.NewAccessControlDashboardPermissionFilter(user, models.PERMISSION_VIEW, searchstore.TypeDashboard).Where()
```

Comment:
basically this PR is this line in the annotation filtering, where we can now use the `user` to filter out the dashboards for the specific user.

most of the other parts is making sure we can pass the `user` object around to this point. Which was a long way I must say 🤣

Reply:
Yup, exactly! 🙈 I should've called it out to make the reviews easier.